staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -113,17 +120,36 @@ type RoundTripperConfig struct {
 	// PingPeriod is a period for sending SPDY Pings on the connection.
 	// Optional.
 	PingPeriod time.Duration
+	// UpgradeTransport is a subtitute transport used for dialing. If set,
+	// this field will be used instead of "TLS" and "Proxier" for connection creation.
+	// Optional.
+	UpgradeTransport http.RoundTripper
 }
 
 // TLSClientConfig implements pkg/util/net.TLSClientConfigHolder for proper TLS checking during
-// proxying with a spdy roundtripper.
+// proxying with a spdy roundtripper. Retrieve tls config from upgradeTransport if it exists.
 func (s *SpdyRoundTripper) TLSClientConfig() *tls.Config {
-	return s.tlsConfig
+	tlsConfig := s.tlsConfig
+	if s.upgradeTransport != nil {
+		var err error
+		tlsConfig, err = utilnet.TLSClientConfig(s.upgradeTransport)
```

Comment:
rather than logging and silently returning the wrong thing here, we can do this extraction in NewRoundTripperWithConfig if UpgradeTransport is non-nil and store the result as `s.tlsConfig`. That'll let you return an error from NewRoundTripperWithConfig if the extraction fails and lets this method stay a simple return

Reply:
Fixed